griptape/tasks/base_text_input_task.py

```diff
@@ -14,14 +15,19 @@ class BaseTextInputTask(BaseTask, ABC):
     DEFAULT_RULESET_NAME = "Default Ruleset"
     ADDITIONAL_RULESET_NAME = "Additional Ruleset"
 
-    input_template: str = field(default=DEFAULT_INPUT_TEMPLATE)
+    input_template: str | TextArtifact | Callable[[BaseTask], TextArtifact] = field(default=DEFAULT_INPUT_TEMPLATE)
     context: dict[str, Any] = field(factory=dict, kw_only=True)
     rulesets: list[Ruleset] = field(factory=list, kw_only=True)
     rules: list[Rule] = field(factory=list, kw_only=True)
 
     @property
     def input(self) -> TextArtifact:
-        return TextArtifact(J2().render_from_string(self.input_template, **self.full_context))
+        if isinstance(self.input_template, TextArtifact):
+            return self.input_template
+        elif isinstance(self.input_template, Callable):
+            return self.input_template(self)
+        else:
+            return TextArtifact(J2().render_from_string(self.input_template, **self.full_context))
```

Comment:
I'm assuming this behaves just fine if the input isn't a template, but I wonder if there's an eventual improvement to be made by typing templates and distinguishing template inputs from raw string inputs, avoiding J2 initialization and template rendering step entirely.

Reply:
If a user doesn't use any of the Jinja syntax, nothing will get rendered. I think the overhead of Jinja unnecessarily rendering is so minimal that it's not worth worrying about.